### PR TITLE
feat: add TS examples to API GQL docs

### PR DESCRIPTION
**Pull request by iartemiev**
#### Description of changes:
Adds TypeScript examples to the API GraphQL category docs.
Sample app updated: https://github.com/aws-amplify/amplify-js-samples-staging/pull/519

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br />
      _ref: MDX: `[link](https://link.com)`
            HTML: `<a href="https://link.com">link</a>`_
            
### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/graphqlapi/js/getting-started.mdx [hunk 1]
```diff
@@ -88,23 +112,74 @@ await API.graphql(graphqlOperation(updateTodo, { input: { id: todoId, name: "Upd
...
+import { OnCreateTodoSubscription } from './API';
+
+// Subscribe to creation of Todo
+const sub = API.graphql<GraphQLSubscription<OnCreateTodoSubscription>>(
+    graphqlOperation(onCreateTodo)
+).subscribe({
+    next: (todoData) => {
+      console.log(todoData);
+      // Do something with the data
+    }
```
Comment (svidgen) [inline src/fragments/lib/graphqlapi/js/getting-started.mdx:166]: This is probably out of scope of this PR, but we may want to take another pass through all of these snippets and ensure we're demonstrating the shape/unpacking of these response objects. `todoData` is a little misleading here, when the actual "`Todo` data" is buried under `todoData.value.data?.onCreateTodo` (or something like that).

**Review by svidgen — DISMISSED** [review]

File: src/fragments/lib/graphqlapi/js/authz.mdx [hunk 2]
```diff
@@ -16,15 +16,38 @@ In order to use this feature with the Amplify GraphQL Client the `API.graphql({.
...
+  variables: { input: todoDetails },
+  authMode: GRAPHQL_AUTH_MODE.AWS_IAM
+});
+```
+
+</Block>
+<Block name="JavaScript">
+
 ```js
+import { GRAPHQL_AUTH_MODE } from '@aws-amplify/api';
```
Comment (renebrandel) [inline src/fragments/lib/graphqlapi/js/authz.mdx:38]: isn't this missing the "API" import?

File: src/fragments/lib/graphqlapi/js/authz.mdx [hunk 3]
```diff
@@ -40,6 +63,28 @@ You can implement your own custom API authorization logic using an AWS Lambda fu
 If you are using a Lambda function as an authorization mode with your AppSync API, you will need to pass an authentication token with each API request and will need to manage token refresh in your application.
 
 The following example assumes `AWS_LAMBDA` is configured as the default authentication type for your API:
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { GraphQLQuery } from '@aws-amplify/api';
```
Comment (renebrandel) [inline src/fragments/lib/graphqlapi/js/authz.mdx:71]: Same comment as above regarding `API` import, and the `queries` import

File: src/fragments/lib/graphqlapi/js/authz.mdx (hunk 2 above)
Comment (iartemiev) [reply src/fragments/lib/graphqlapi/js/authz.mdx:38]: It's supposed to be more of a diff from the last example... But if you think it makes things clearer, I can add all the imports used by the snippet

File: src/fragments/lib/graphqlapi/js/getting-started.mdx (hunk 1 above)
Comment (iartemiev) [reply src/fragments/lib/graphqlapi/js/getting-started.mdx:166]: I was debating whether to unwrap it or not. I thought the entire subscription payload may be of value to some customers, but perhaps at least a better param name here would clear things up

**Review by renebrandel — APPROVED** [review]

File: src/fragments/lib/graphqlapi/js/getting-started.mdx [hunk 4]
```diff
@@ -160,9 +160,9 @@ import { OnCreateTodoSubscription } from './API';
 const sub = API.graphql<GraphQLSubscription<OnCreateTodoSubscription>>(
     graphqlOperation(onCreateTodo)
 ).subscribe({
-    next: (todoData) => {
-      console.log(todoData);
-      // Do something with the data
+    next: (payload) => {
+      const createdTodo = payload.value.data?.onCreateTodo;
+      console.log(createdTodo);
```
Comment (svidgen) [inline src/fragments/lib/graphqlapi/js/getting-started.mdx:165]: Nice! Thank-you!

File: src/fragments/lib/graphqlapi/js/authz.mdx [hunk 5]
```diff
@@ -40,27 +65,82 @@ You can implement your own custom API authorization logic using an AWS Lambda fu
 If you are using a Lambda function as an authorization mode with your AppSync API, you will need to pass an authentication token with each API request and will need to manage token refresh in your application.
 
 The following example assumes `AWS_LAMBDA` is configured as the default authentication type for your API:
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+// ...
```
Comment (svidgen) [inline src/fragments/lib/graphqlapi/js/authz.mdx:73]: I see a few of these ellipses "headers" were added in the latest rev as if to acknowledge "all the normal imports and configs happen here". But, I'm not sure I see the need. The rest of our docs seem content to assume imports and context are implied in each example. Is there a reason we're breaking with that pattern in this PR?

**Review by svidgen — APPROVED** [review]

File: src/fragments/lib/graphqlapi/js/authz.mdx (hunk 5 above)
Comment (iartemiev) [reply src/fragments/lib/graphqlapi/js/authz.mdx:73]: Chatted about this offline. Resolving

**Review by stocaaro — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**